api/capo/v1alpha4/openstack_cluster_types.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// OpenStackCluster is the Schema for the openstackclusters API.
+type OpenStackCluster struct {
```

Comment:
I'd like to get @giantswarm/sig-operator's thoughts on this approach. We discussed in our sync a few months ago that this is one possible way to avoid importing large projects such as `cluster-api-provider-openstack`. The general idea is to define a minimal struct with only the fields that are needed (e.g. `.status.network.id`). By adding this to the group and registering it in the scheme used by the k8s client, we can `Get` and `List` this type without adding an external import. @kopiczko I think you said you had used this before, right?

Reply:
It seems like this is the suggested approach by upstream.